pkg/kubernetes/deployment/v1alpha1/kubernetes.go

```diff
@@ -0,0 +1,199 @@
+package v1alpha1
+
+import (
+	kclient "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	api_apps_v1 "k8s.io/api/apps/v1"
+	api_extn_v1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// getClientsetFn is a typed function that abstracts fetching of internal clientset
+type getClientsetFn func() (clientset *kubernetes.Clientset, err error)
+
+// getExtnV1Beta1Fn is a typed function that abstracts get of deployment instances of ExtnV1Beta1 api group
+type getExtnV1Beta1Fn func(cli *kubernetes.Clientset, name, namespace string) (*api_extn_v1beta1.Deployment, error)
+
+// getAppsV1Fn is a typed function that abstracts get of deployment instances of appsv1 api group
+type getAppsV1Fn func(cli *kubernetes.Clientset, name, namespace string) (*api_apps_v1.Deployment, error)
+
+// rollOutStatusExtnV1Beta1Fn is a typed function that abstracts rollout status
+//  of deployment instances of ExtnV1Beta1 api group
+type rollOutStatusExtnV1Beta1Fn func(cli *kubernetes.Clientset, name, namespace string) ([]byte, error)
+
+// rollOutStatusAppsV1Fn is a typed function that abstracts rollout status
+//  of deployment instances of AppsV1 api group
+type rollOutStatusAppsV1Fn func(cli *kubernetes.Clientset, name, namespace string) ([]byte, error)
+
+// kubeclient enables kubernetes API operations on deployment instance
+type kubeclient struct {
+	// clientset refers to kubernetes clientset. It is responsible to
+	// make kubernetes API calls for crud op
+	clientset *kubernetes.Clientset
+
+	// functions useful during mocking
+	getClientset getClientsetFn
+
+	getExtnV1Beta1 getExtnV1Beta1Fn
+
+	getAppsV1 getAppsV1Fn
+
+	rollOutStatusExtnV1Beta1 rollOutStatusExtnV1Beta1Fn
+
+	rollOutStatusAppsV1 rollOutStatusAppsV1Fn
+}
+
+// rolloutOutput struct contaons message and boolean value to show rolloutstatus
+type rolloutOutput struct {
+	IsRolledout bool   `json:"IsRolledout"`
```

Comment:
Let us keep the json property in camelCase i.e. `json:"isRolledout"`

Reply:
done